src/nvim/api/extmark.c

```diff
@@ -203,27 +203,9 @@ ArrayOf(Integer) nvim_buf_get_extmark_by_id(Buffer buffer, Integer ns_id,
     return rv;
   });
 
-  bool details = false;
-  bool hl_name = true;
-  for (size_t i = 0; i < opts.size; i++) {
-    String k = opts.items[i].key;
-    Object *v = &opts.items[i].value;
-    if (strequal("details", k.data)) {
-      details = api_object_to_bool(*v, "details", false, err);
-      if (ERROR_SET(err)) {
-        return rv;
-      }
-    } else if (strequal("hl_name", k.data)) {
-      hl_name = api_object_to_bool(*v, "hl_name", false, err);
-      if (ERROR_SET(err)) {
-        return rv;
-      }
-    } else {
-      VALIDATE_S(false, "'opts' key", k.data, {
-        return rv;
-      });
-    }
-  }
+  bool details = HAS_KEY(opts, get_extmark, details) ? opts->details : false;
```

Comment:
```suggestion
  bool details = opts->details;
```

Reply:
Fixed in af08794ed29ae67c508d54e9d97b5341daec2892.